sklearn/feature_extraction/tests/test_image.py

```diff
@@ -170,6 +171,26 @@ def test_extract_patches_max_patches():
                   max_patches=-1.0)
 
 
+def test_extract_patch_same_size_image():
+    face = downsampled_face
+    i_h, i_w = face.shape
+
+    # Request patches of the same size as image
+    # Should return just the single patch a.k.a. the image
+    patches = extract_patches_2d(face, face.shape, max_patches=2)
+    assert_equal(patches.shape, (1, i_h, i_w))
+
+
+def test_extract_patches_less_than_max_patches():
+    face = downsampled_face
+    i_h, i_w = face.shape
+    p_h, p_w = int(3*i_h/4), int(3*i_w/4)  # get patches of 3/4th the image size
```

Comment:
Can you put some space between the mathematical operator and you don't need for any comment. This is straightforward

Reply:
You should also used python 3 standard without casting

```python
p_h, p_w = 3 * i_h // 4, 3 * i_w // 4
```

Reply:
Just add at the beginning of the file 

```python
from __future__ import division
```

Reply:
Awesome! I was trying to be backwards compatible. Thank you!